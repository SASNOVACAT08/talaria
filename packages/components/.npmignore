node_modules/
src/
types/
.eslintignore
.eslintrc
tsconfig.json
tsconfig.node.json
vite.config.ts