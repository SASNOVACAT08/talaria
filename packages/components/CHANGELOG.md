# @tlra/components

## 0.2.0

### Minor Changes

- add preview & hr

## 0.1.0

### Patch Changes

- 6c66eb9: first version
