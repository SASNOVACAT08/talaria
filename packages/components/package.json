{
  "name": "@tlra/components",
  "private": false,
  "version": "0.2.0",
  "type": "module",
  "files": [
    "dist"
  ],
  "main": "./dist/components.umd.js",
  "module": "./dist/components.es.js",
  "exports": {
    ".": {
      "import": "./dist/components.es.js",
      "require": "./dist/components.umd.js"
    }
  },
  "types": "./dist/index.d.ts",
  "scripts": {
    "build": "vue-tsc && vite build",
    "lint": "eslint --ext .ts,.vue .",
    "lint:fix": "eslint --ext .ts,.vue . --fix"
  },
  "dependencies": {
    "vue": "^3.2.45"
  },
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^5.48.0",
    "@typescript-eslint/parser": "^5.48.0",
    "@vitejs/plugin-vue": "^4.0.0",
    "@vue/eslint-config-prettier": "^7.0.0",
    "@vue/eslint-config-typescript": "^11.0.2",
    "eslint": "^8.31.0",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-vue": "^9.8.0",
    "prettier": "^2.6.2",
    "rollup-plugin-typescript2": "^0.34.1",
    "typescript": "^4.9.4",
    "vite": "^4.0.0",
    "vue-tsc": "^1.0.11"
  }
}
