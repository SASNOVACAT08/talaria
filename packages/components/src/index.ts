import Html from './Html.vue';
import Body from './Body.vue';
import Link from './Link.vue';
import Container from './Container.vue';

export { Html, Body, Link, Container };
