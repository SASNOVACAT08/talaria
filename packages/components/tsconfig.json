{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "lib": ["ESNext", "DOM"],
    "declaration": true
  },
  "include": ["src/**/*.ts", "src/**/*.d.ts", "src/**/*.tsx", "src/**/*.vue", "vite-env.d.ts", "types/**/*.d.ts"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
