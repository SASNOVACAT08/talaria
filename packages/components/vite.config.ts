import { defineConfig } from 'vite';
import vue from '@vitejs/plugin-vue';
import typescript2 from 'rollup-plugin-typescript2';

export default defineConfig({
  build: {
    lib: {
      entry: 'src/index.ts',
      name: 'components',
      fileName: (format) => `components.${format}.js`,
    },
    rollupOptions: {
      external: ['vue'],
      output: {
        globals: {
          vue: 'Vue',
        },
      },
    },
  },
  plugins: [
    vue(),
    typescript2({
      check: false,
      include: ['src/**/*.ts', 'src/**/*.vue'],
    }),
  ],
});
