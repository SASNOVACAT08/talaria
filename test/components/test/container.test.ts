import { describe, it, expect } from 'vitest';
import { Container } from '@tlra/components';
import { mount } from '@vue/test-utils';

describe('Container.vue', () => {
  it('should render by default', () => {
    const wrapper = mount(Container);
    const html = wrapper.html().trim().replace(/\n/g, '');
    expect(html).toMatchSnapshot();
  });

  it('should render with <p>', () => {
    const wrapper = mount(Container, {
      slots: {
        default: '<p>Hello World</p>',
      },
    });
    expect(wrapper.find('p').text()).toContain('Hello World');
  });

  it('should render with custom style', () => {
    const wrapper = mount(Container, {
      props: {
        style: {
          color: 'red',
        },
      },
    });
    const html = wrapper.html().trim().replace(/\n/g, '');
    expect(html).toMatchSnapshot();
  });
});
