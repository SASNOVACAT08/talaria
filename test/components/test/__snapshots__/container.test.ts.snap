// Vitest Snapshot v1

exports[`Container.vue > should render by default 1`] = `"<div v-html=\\"&lt;!--[if mso | IE]&gt;&lt;table role=&#x27;presentation&#x27; width=&#x27;100%&#x27; align=&#x27;center&#x27; style=maxWidth:37.5em;&gt;&lt;tr&gt;&lt;td&gt;&lt;/td&gt;&lt;td style=&#x27;width:37.5em;background:#ffffff&#x27;&gt;&lt;![endif]--&gt;\\"></div><div style=\\"max-width: 37.5em;\\"></div><div v-html=\\"&lt;!--[if mso | IE]&gt;&lt;/td&gt;&lt;td&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;![endif]--&gt;\\"></div>"`;

exports[`Container.vue > should render with custom style 1`] = `"<div v-html=\\"&lt;!--[if mso | IE]&gt;&lt;table role=&#x27;presentation&#x27; width=&#x27;100%&#x27; align=&#x27;center&#x27; style=maxWidth:37.5em;color:red;&gt;&lt;tr&gt;&lt;td&gt;&lt;/td&gt;&lt;td style=&#x27;width:37.5em;background:#ffffff&#x27;&gt;&lt;![endif]--&gt;\\"></div><div style=\\"max-width: 37.5em; color: red;\\"></div><div v-html=\\"&lt;!--[if mso | IE]&gt;&lt;/td&gt;&lt;td&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;![endif]--&gt;\\"></div>"`;
